### PR TITLE
feat: Design System

### PR DESCRIPTION
**Pull request by cecipirotto**
Add buttons and text fields to design system
Add catalog

https://github.com/xmartlabs/supabase-ws-flutter/assets/31603856/5da73079-e347-4a38-924d-59f7b852b272

### CODE REVIEW

File: supabase-ws-flutter-mobile/lib/catalog_main.dart [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+              child: ListView(
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.symmetric(vertical: 6.0),
+                    child: AppPrimaryButton(
+                      text: "Filled Button",
+                      onPressed: () {
+                        if (kDebugMode) {
+                          print("Filled button pressed!");
+                        }
```
Comment (mirland) [inline supabase-ws-flutter-mobile/lib/catalog_main.dart:57]: toast?

File: supabase-ws-flutter-mobile/lib/catalog_main.dart [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    ),
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(vertical: 6.0),
+                    child: AppPrimaryButton(
+                      text: "Stroke Button",
+                      onPressed: () {
+                        if (kDebugMode) {
+                          print("Stroke button pressed!");
+                        }
```
Comment (mirland) [inline supabase-ws-flutter-mobile/lib/catalog_main.dart:69]: I'd remove the debug condition and I'd use toast or something like that

File: supabase-ws-flutter-mobile/lib/ui/widgets/design_system/buttons/primary_button.dart [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  Widget build(BuildContext context) {
+    switch (style) {
+      case StyleButton.filled:
+        return _primaryFilledButton(context);
+      case StyleButton.stroke:
+        return _primaryStrokeButton(context);
+      case StyleButton.ghost:
+        return _primaryGhostButton(context);
+    }
+  }
```
Comment (mirland) [inline supabase-ws-flutter-mobile/lib/ui/widgets/design_system/buttons/primary_button.dart:74]: I'd not use methods here, but he have to discuss it

File: supabase-ws-flutter-mobile/lib/ui/widgets/design_system/text_fields/input_text.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_template/ui/extensions/context_extensions.dart';
+import 'package:flutter_template/ui/theme/app_theme.dart';
+
+class InputText extends StatelessWidget {
```
Comment (mirland) [inline supabase-ws-flutter-mobile/lib/ui/widgets/design_system/text_fields/input_text.dart:6]: AppTextInputField? Just to be consistent

**Review by mirland — COMMENTED** [review]
Add an image preview in the pr please 🙏

File: supabase-ws-flutter-mobile/lib/ui/widgets/design_system/text_fields/input_text.dart [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            borderRadius: BorderRadius.circular(4.r),
+          ),
+          focusedErrorBorder: OutlineInputBorder(
+            borderSide: BorderSide(color: context.theme.colors.error.shade400),
+            borderRadius: BorderRadius.circular(4.r),
+          ),
+          prefixIcon: Icon(leadingIcon),
+          labelText: labelText,
+          labelStyle: context.theme.textStyles.bodySmall?.copyWith(
+            color: context.theme.colors.onSurface,
```
Comment (nicolantean) [inline supabase-ws-flutter-mobile/lib/ui/widgets/design_system/text_fields/input_text.dart:60]: I think this is not the color used in the DS

File: supabase-ws-flutter-mobile/lib/ui/widgets/design_system/text_fields/input_text.dart [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+          hintText: hintText,
+          hintStyle: context.theme.textStyles.bodySmall?.copyWith(
+            color: context.theme.colors.textColor.shade300,
+          ),
+          errorStyle: context.theme.textStyles.bodySmall?.copyWith(
+            color: context.theme.colors.error.shade300,
+          ),
+        ),
+        style: context.theme.textStyles.bodySmall?.copyWith(
+          color: context.theme.colors.onSurface,
```
Comment (nicolantean) [inline supabase-ws-flutter-mobile/lib/ui/widgets/design_system/text_fields/input_text.dart:71]: same here

**Review by nicolantean — APPROVED** [review]

**Pull request merged**